textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcherTest.java

```diff
@@ -0,0 +1,217 @@
+        double threshold = 0.5; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+        List<ITuple> results = getQueryResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithThresholdVariation() throws Exception {
+        String query = "Twelve Angry Men Cafe";
+        double threshold = 0.25;
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+ 
+        Attribute[] schemaAttributes = TestConstants.ATTRIBUTES_PEOPLE;
+        
+        IField[] fields1 = { new StringField("bruce"), new StringField("john Lee"), new IntegerField(46),
+                new DoubleField(5.50), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-14-1970")),
+                new TextField("Tall Angry") };
+        IField[] fields2 = { new StringField("brad lie angelina"), new StringField("pitt"), new IntegerField(44),
+                new DoubleField(6.10), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-12-1972")),
+                new TextField("White Angry") };
+        IField[] fields3 = { new StringField("george lin lin"), new StringField("lin clooney"), new IntegerField(43),
+                new DoubleField(6.06), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-13-1973")),
+                new TextField("Lin Clooney is Short and lin clooney is Angry") };
+        
+        ITuple tuple1 = new DataTuple(new Schema(schemaAttributes), fields1);
+        ITuple tuple2 = new DataTuple(new Schema(schemaAttributes), fields2);
+        ITuple tuple3 = new DataTuple(new Schema(schemaAttributes), fields3);
+        List<ITuple> expectedResultList = new ArrayList<>();
+        expectedResultList.add(tuple1);
+        expectedResultList.add(tuple2);
+        expectedResultList.add(tuple3);
+        
+        List<ITuple> results = getQueryResults(query, threshold, attributeList, getSpan);
+        boolean contains = TestUtils.containsAllResults(expectedResultList, results);
+        //Assert.assertTrue(contains);
+        //Assert.assertEquals(3,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherForStringField() throws Exception {
+        String query = "tom hanks";
+        double threshold = 1; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.FIRST_NAME_ATTR);
+        List<ITuple> results = getQueryResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithoutSpan() throws Exception {
```

Comment:
It will be good to add one more test case with a long text field (say, more than 50 tokens) and see if the operator works properly.